test/tools/yulopti.cpp

```diff
@@ -214,47 +338,40 @@ class YulOpti
 					case '#':
 						return;
 					case ',':
-						VarNameCleaner::run(m_context, *m_ast);
+						runVarNameCleaner();
 						// VarNameCleaner destroys the unique names guarantee of the disambiguator.
 						disambiguated = false;
 						break;
 					case ';':
-					{
-						Object obj;
-						obj.code = m_ast;
-						StackCompressor::run(m_dialect, obj, true, 16);
+						runStackCompressor();
 						break;
-					}
 					default:
-						OptimiserSuite{m_context}.runSequence(
-							std::string_view(&option, 1),
-							*m_ast
-						);
+						runSequence(string_view(&option, 1));
 				}
-				_source = AsmPrinter{m_dialect}(*m_ast);
+
+				resetNameDispenser();
+				runCodeAnalyzer(errorReporter);
```

Comment:
Rerunning analysis after every step is probably superfluous. Or at least, `yulopti` was not doing that before. Any reason to introduce it now?

Reply:
I removed it since I didn't see a reason for it there either.